web/src/integTest/java/com/microsoft/applicationinsights/azuretest/CompareTelemetryWithAzureExportTests.java

```diff
@@ -0,0 +1,342 @@
+
+        ArrayList<TelemetryItem> expectedTelemetries = new ArrayList<TelemetryItem>();
+        expectedTelemetries.add(expectedTelemetry1);
+        expectedTelemetries.add(expectedTelemetry2);
+        expectedTelemetries.add(expectedTelemetry3);
+
+        return expectedTelemetries;
+    }
+
+    /**
+     * Sends HTTP GET request and returns the expected telemetry from AppInsights
+     * @param uri The URI for the request
+     * @return The expected telemetry items from AppInsights
+     * @throws Exception
+     */
+    private TelemetryItem sendGet(URI uri, String requestId) throws Exception {
+        HttpURLConnection connection = (HttpURLConnection) uri.toURL().openConnection();
+
+        // optional default is GET
+        connection.setRequestMethod("GET");
+
+        //add request header
+        connection.setRequestProperty("User-Agent", config.getUserAgent());
+
+        System.out.println("Sending 'GET' request to URL: " + uri.toString());
+        Integer responseCode = connection.getResponseCode();
+        System.out.println("Response Code : " + responseCode);
+
+        // Create expected result
+        TelemetryItem expectedResult = new TelemetryItem(DocumentType.Requests, requestId);
+        expectedResult.setProperty("port", Integer.toString(uri.getPort()));
+        expectedResult.setProperty("responseCode", responseCode.toString());
+        expectedResult.setProperty("uri", uri.toString());
+        expectedResult.setProperty("userAgent", config.getUserAgent());
+
+        try {
+            String[] params = uri.getQuery().split("&");
+            for (String param : params) {
+                String name = param.split("=")[0];
+                String value = param.split("=")[1];
+                expectedResult.setProperty("queryParameter." + name, value);
+            }
+        }
+        catch (Exception e) {
+            System.out.println("Failed to process query parameters '" + uri.getQuery() + "'. Ignoring.");
+        }
+
+        return expectedResult;
+    }
+
+    /**
+     * Connects to Azure export and retrieves the telemetry items
+     * @param docType The document type to retrieve from Azure
+     * @return The retrieved telemetry items
+     * @throws Exception
+     */
+    private ArrayList<TelemetryItem> getTelemetryFromAzure(DocumentType docType) throws Exception {
+        System.out.println("Creating Azure storage account connection");
+        CloudStorageAccount account = CloudStorageAccount.parse(config.getStorageConnectionString());
+        CloudBlobClient blobClient = account.createCloudBlobClient();
+
+        System.out.println("Creating Azure queue connection");
+        CloudQueueClient queueClient = account.createCloudQueueClient();
+        CloudQueue queue = queueClient.getQueueReference(config.getQueueName());
+
+        ArrayList<JSONObject> telemetryAsJson = new ArrayList<JSONObject>();
+        StopWatch stopWatch = new StopWatch();
```

Comment:
I think it's more elegant this way than using "new Date().getTime() - startTime.getTime() < timeToPoll ...."